### PR TITLE
Generate symbols file for test plan

### PR DESCRIPTION
**Pull request by nashif**
- build: generate symbols file
- ztest: generate symbols for tests
- cmake: binutils: find binutils nm for host compiler

### CODE REVIEW

File: Kconfig.zephyr [hunk 1]
```diff
@@ -377,6 +377,12 @@ config OUTPUT_STAT
 	help
 	  Create a stat file using readelf -e <elf>
 
+config OUTPUT_SYMBOLS
+	bool "Create a symbol file"
+	default n
```
Comment (stephanosio) [inline Kconfig.zephyr:382]: `default n` is implied and redundant. It should be removed.

**Comment by stephanosio** [comment]
@nashif I assume, with this, we will no longer need an abomination like the following, is that correct?

https://github.com/zephyrproject-rtos/zephyr/blob/735ddeb4a8e26c6ce07d6aa5a0b77fffc559b2a8/tests/lib/newlib/thread_safety/testcase.yaml#L7-L14

**Comment by nashif** [comment]
> @nashif I assume, with this, we will no longer need an abomination like the following, is that correct?

yes

**Review by stephanosio — DISMISSED** [review]

File: cmake/bintools/gnu/target_bintools.cmake [hunk 2]
```diff
@@ -124,3 +124,10 @@ set_property(TARGET bintools PROPERTY readelf_flag_outfile ">;" )
 #set_property(TARGET bintools PROPERTY readelf_flag_headers -E)
 #set_property(TARGET bintools PROPERTY readelf_flag_infile "")
 #set_property(TARGET bintools PROPERTY readelf_flag_outfile "-O file=" )
+
+
+set_property(TARGET bintools PROPERTY nm_command ${CMAKE_NM})
```
Comment (tejlmand) [inline cmake/bintools/gnu/target_bintools.cmake:129]: could you come up with a more generic name here ?
Like is done for `disassembly`, `strip`, `elfconvert`

`nm` is very GNU centric, and other toolchains may provide other commands and arguments for achieving similar functionality.
So would be nice with more generic name on what you want to achieve.

**Review by tejlmand — CHANGES_REQUESTED** [review]
Please add this new bintools command property to this template file:
https://github.com/zephyrproject-rtos/zephyr/blob/55df2bf08ff0bb9abbc46b11d252d6b310a604c9/cmake/bintools/bintools_template.cmake#L7-L13

so that users using other toolchains won't get an error when enabling this.

Also, could we find a more generic name, that are then mapped by toolchain implementations.
For example: `symbol_command` or `symbol_list_command` ?

File: cmake/bintools/bintools_template.cmake [hunk 3]
```diff
@@ -108,6 +109,13 @@
 #   strip_flag_dwo     : Flag for removing dwarf sections
 #   strip_flag_infile  : Flag for specifying the input file
 #   strip_flag_outfile : Flag for specifying the output file
+#
+# - symbols : Name of command for printing out symbols
+#             For GNU binajry utilities this is nm
```
Comment (yperess) [inline cmake/bintools/bintools_template.cmake:114]: nit `s/binajry/binary`

**Review by yperess — COMMENTED** [review]
LGTM, one small nit on a mispelled word

**Review by stephanosio — APPROVED** [review]

**Review by tejlmand — APPROVED** [review]
lgtm.

A minor alignment issuse noticed, but nothing blocking.

**Pull request merged**